pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -276,6 +280,11 @@ func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMoun
 					volumeMounts = append(volumeMounts, volumeMountsTLS)
 				}
 				envVars = append(envVars, kms.ConfigToEnvVar(c.spec)...)
+				if c.spec.Security.KeyManagementService.IsKMIPKMS() {
+					envVars = append(envVars, cephVolumeRawEncryptedEnvVarFromSecret(osdProps))
+					_, volmeMountsKMIP := kms.KMIPVolumeAndMount(c.spec.Security.KeyManagementService.TokenSecretName)
```

Comment:
@Rakshith-R do we need the volume struct and mount structure validated in the return path?

Reply:
> @Rakshith-R do we need the volume struct and mount structure validated in the return path?

I'll add some unit test cases for that function in a while.